### PR TITLE
Fix mislooping in ssh.go. Add retries to AddSSHKeys.

### PR DESCRIPTION
**Pull request by cjcullen**
@brendandburns @roberthbailey

### CODE REVIEW

File: pkg/cloudprovider/gce/gce.go [hunk 1]
```diff
@@ -483,37 +484,47 @@ func (gce *GCECloud) getInstanceByName(name string) (*compute.Instance, error) {
...
+					Key:   "sshKeys",
+					Value: keyString,
+				})
+		}
+		op, err := gce.service.Projects.SetCommonInstanceMetadata(gce.projectID, project.CommonInstanceMetadata).Do()
+		if err != nil {
+			glog.Errorf("Could not get Set Metadata: %v", err)
+			return false, nil
+		}
+		err = gce.waitForGlobalOp(op)
```
Comment (roberthbailey) [inline pkg/cloudprovider/gce/gce.go]: nit: if err := gce.waitForGlobalOp(op); err != nil { ... }

**Comment by roberthbailey** [comment]
LGTM.

File: pkg/cloudprovider/gce/gce.go (hunk 1 above)
Comment (cjcullen) [inline pkg/cloudprovider/gce/gce.go]: Done.

**Comment by cjcullen** [comment]
I also bumped up the AddSSHKey timeout to 30 seconds.

**Comment by cjcullen** [comment]
I spun up a 15 node GKE cluster and verified that the retries work as intended.

**Comment by roberthbailey** [comment]
Reassigning to @brendandburns since he had more comments on the code than I did.

**Comment by brendandburns** [comment]
LGTM

**Comment by brendandburns** [comment]
mergez

**Pull request merged**